testsuite/features/support/quality_intelligence.rb

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) 2024 SUSE LLC.
+# Licensed under the terms of the MIT license.
+require_relative 'database_handler'
+require_relative 'metrics_collector_handler'
+
+# Quality Intelligence handler to produce, parse and report Quality Intelligence from the test suite
+class QualityIntelligence
+  QI = 'quality_intelligence'.freeze
+  private_constant :QI
+
+  # Initialize the QualityIntelligence handler
+  def initialize
+    @db_handler = DatabaseHandler.new(ENV.fetch('REDIS_HOST', nil), ENV.fetch('REDIS_PORT', nil), ENV.fetch('REDIS_USERNAME', nil), ENV.fetch('REDIS_PASSWORD', nil))
+    @metrics_collector_handler = MetricsCollectorHandler.new(ENV.fetch('PROMETHEUS_PUSH_GATEWAY_URL', 'http://nsa.mgr.suse.de:9091'))
```

Comment:
What happens if Prometheus/Database cannot be contacted, not even if the URL is bad, but when it's after firewall/cert validation error etc?

Reply:
It gonna raise an error and the cucumber process will fail, as expected.
I mean, we don't want to keep running under such conditions.

Reply:
My concert is that errors that will raise may be unclear, like connection timeout error, but it's likely to be managable.

Closing